From 390c3ee882dcb0bd9241fa4ef4a1c7537cea9796 Mon Sep 17 00:00:00 2001
From: Anjaliintel-21 <anjali.ray@intel.com>
Date: Mon, 7 Aug 2023 11:11:58 +0000
Subject: [PATCH] Add PowerCapacityWatts instead on PowerInputwatts

As PowerInputwatts and PowerOutputwatts are not present in xml file,
so we are replacing PowerInputwatts to PowerCapacityWatts and removing
PowerOutputwatts.

Tested with the older version with
/redfish/v1/Chassis/BNC_Baseboard/PowerSubsystem/PowerSupplies is not
null.
Validator passed with this URI:
`/redfish/v1/Chassis/BNC_Baseboard/PowerSubsystem/PowerSupplies/SOLUM_CO_IS162F22_PSU1

Signed-off-by: Anjaliintel-21 <anjali.ray@intel.com>
---
 redfish-core/lib/sensors.hpp | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/redfish-core/lib/sensors.hpp b/redfish-core/lib/sensors.hpp
index 13880cb7..c1b89288 100644
--- a/redfish-core/lib/sensors.hpp
+++ b/redfish-core/lib/sensors.hpp
@@ -852,11 +852,11 @@ inline void objectPropertiesToJson(
         }
         else if (!(boost::ifind_first(sensorName, "input").empty()))
         {
-            unit = "/PowerInputWatts"_json_pointer;
+            unit = "/PowerCapacityWatts"_json_pointer;
         }
-        else
+        else if (!(boost::ifind_first(sensorName, "output").empty()))
         {
-            unit = "/PowerOutputWatts"_json_pointer;
+            return;
         }
     }
     else
-- 
2.17.1

